cdc/model/sink.go

```diff
@@ -719,6 +748,115 @@ func (t *SingleTableTxn) GetCommitTs() uint64 {
 	return t.CommitTs
 }
 
+// TrySplitAndSortUpdateEvent split update events if unique key is updated
+func (t *SingleTableTxn) TrySplitAndSortUpdateEvent() error {
+	if len(t.Rows) < 2 {
+		return nil
+	}
+	newRows, err := convertRowChangedEvents(t.Rows)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	sort.Sort(RowChangedEvents(newRows))
+	t.Rows = newRows
+	return nil
+}
+
+// convertRowChangedEvents uses to convert RowChangedEvents to TableSinkRowChangedEvents.
+// It will deal with the old value compatibility.
+func convertRowChangedEvents(
+	events []*RowChangedEvent,
+) ([]*RowChangedEvent, error) {
+	rowChangedEvents := make([]*RowChangedEvent, 0, len(events))
+	for _, e := range events {
+		if e == nil {
+			log.Warn("skip emit nil event",
+				zap.Any("event", e))
+			continue
+		}
+
+		colLen := len(e.Columns)
+		preColLen := len(e.PreColumns)
+		// Some transactions could generate empty row change event, such as
+		// begin; insert into t (id) values (1); delete from t where id=1; commit;
+		// Just ignore these row changed events.
+		if colLen == 0 && preColLen == 0 {
+			log.Warn("skip emit empty row event",
+				zap.Any("event", e))
+			continue
+		}
+
+		// This indicates that it is an update event,
+		// and after enable old value internally by default(but disable in the configuration).
```

Comment:
do you mean that `enable-old-value` is always false in all conditions ?

Reply:
yes,  we should always split the update events if unique key is updated